spec/fluent.ebnf

```diff
@@ -18,7 +18,13 @@ Term                ::= "-" Identifier blank_inline? "=" blank_inline? Value Att
  * the AST construction. */
 CommentLine         ::= ("###" | "##" | "#") ("\u0020" /.*/)? line_end
 
-/* Adjacent junk_lines are joined into FTL.Junk during the AST construction. */
+/* Junk represents unparsed content.
+ *
+ * Junk is parsed line-by-line until a line is found which looks like it might
+ * be a beginning of a new message, term, or a comment. Any whitespace
+ * following a broken Entry is also considered part of Junk.
+ */
+Junk                ::= junk_line (junk_line - "#" - "-" - [a-zA-Z])*
```

Comment:
This rendering to ebnf doesn't make any sense, right?

Reply:
I think it does. We're using the EBNF syntax as defined in the [XML spec](https://www.w3.org/TR/xml/#sec-notation), with an extension of allowing regexes in a few places. The XML one reads:

     A - B
        matches any string that matches A but does not match B.

So `junk_line - "#"` matches a line of junk which doesn't start with a `#`. I think that's exactly what we want to say here.

Reply:
> # foo\n does match junkline, but it doesn't match #, so it's a junk line.

Can you rephrase this please?

To me, the EBNF in this PR clearly expresses the intent. This is already a slippery slope because we're trying to define how to parse unparsed content. I don't want to overthink it. I'm also not sure how you'd like to write this differently. I could look at a PR if you'd like to prepare one :)

Reply:
I just took the literal quote from the xml spec, and replaced `A` and `B` with `junk_line` and `#`, resp. And tested that against a candidate line line `# foo\n`.

I know that you don't believe in the value of the EBNF, and I don't care much either about this one.

Reply:
> I just took the literal quote from the xml spec, and replaced A and B with junk_line and #, resp. And tested that against a candidate line line # foo\n.

Ah, I see what you mean, thanks. `#` matches `# foo\n` partially and that's enough for a negative lookahead to work here. I guess we could try to refactor this into something like `sequence(and(not("#"), any_char), junk_line)` but it would require special handling of blank lines inside of junk. (`any_char` doesn't parse newlines.) All in all, I favor the expressiveness of the approach I implemented in this PR.